persistence/src/main/java/edu/unc/lib/dl/persist/services/deposit/DepositModelManager.java

```diff
@@ -0,0 +1,479 @@
+        }
+    }
+
+    /**
+     * Perform a sparql query against the deposit model
+     *
+     * @param depositPid pid of the deposit
+     * @param queryString sparql query
+     * @return results of the query, serialized as csv in an output stream
+     * @throws IOException
+     */
+    public String performQuery(PID depositPid, String queryString) throws IOException {
+        Model depositModel = getReadModel(depositPid);
+
+        Query query = QueryFactory.create(queryString);
+
+        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+
+        try (
+                QueryExecution qexec = QueryExecutionFactory.create(query, depositModel);
+                Writer writer = new PrintWriter(outStream);
+                CSVPrinter printer = new CSVPrinter(writer, CSVFormat.DEFAULT);
+                ) {
+            ResultSet results = qexec.execSelect();
+            List<String> varNames = results.getResultVars();
+
+            while (results.hasNext()) {
+                QuerySolution soln = results.nextSolution();
+
+                for (String varName : varNames) {
+                    printer.print(soln.get(varName));
+                }
+                printer.println();
+            }
+        }
+
+        return outStream.toString("UTF-8");
+    }
+
+    /**
+     * Commit changes to the dataset
+     * @param depositPid pid of the deposit
+     * @param dataset dataset to commit, or null, in which case the dataset will be retrieved
+     * @param endTx end the transaction if true
+     */
+    public void commit(PID depositPid, Dataset dataset, boolean endTx) {
+        if (dataset == null) {
+            commit(depositPid, endTx);
+        } else {
+            commit(dataset, endTx);
+        }
+    }
+
+    /**
+     * Commit changes to the dataset for the specified deposit
+     * @param depositPid pid of the deposit
+     * @param endTx end the transaction if true
+     */
+    public void commit(PID depositPid, boolean endTx) {
+        String depositUri = depositPid.getURI();
+
+        Lock lock = depositLocker.get(depositUri);
+        try {
+            lock.lockInterruptibly();
+
+            Dataset dataset = loadDataset(depositPid);
+            commit(dataset, endTx);
+        } catch (InterruptedException e) {
+            throw new InterruptedRuntimeException(e);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private void commit(Dataset dataset, boolean endTx) {
+        if (dataset.isInTransaction()) {
+            dataset.commit();
+            if (endTx) {
+                dataset.end();
+            }
+        }
+    }
+
+    /**
+     * Commit or abort changes in the dataset
+     * @param depositPid pid of the deposit
+     * @param dataset dataset to commit, or null, in which case the dataset will be retrieved
+     * @param abort if true, the commit will be aborted
+     */
+    public void commitOrAbort(PID depositPid, Dataset dataset, boolean abort) {
```

Comment:
Should this variant of the method not worry about checking if the dataset is null and just assume it is? Then there's no need to pass it in "null".